### ISSUE TITLE
Slow inference(?) despite `torch.compile` and `fuse_loras` for SDXL

### ISSUE THREAD

**Issue by RizzwareEngineer** [huggingface/diffusers#7357]
### Describe the bug

I am attempting to fine-tune an SDXL model by fusing a few LoRA adapters, then utilizing `torch.compile` per this portion of the [Merge LoRAs documentation](https://huggingface.co/docs/diffusers/main/en/using-diffusers/merge_loras#torchcompile) to generate images of my desired style while improving the SDXL's inference speed and reduce memory-usage. [My Colab notebook](https://colab.research.google.com/drive/14JTBSkKKkvr8BwUb9CH59ZtLGwzprE2T?usp=sharing) also takes from ["Accelerate inference of text-to-image diffusion models"](https://huggingface.co/docs/diffusers/main/en/tutorials/fast_diffusion).

I am simply concerned as to why the first image generation run takes (for me, ~15 minutes on avg.) long. I thought the `Compile UNet` Colab cell was supposed to be the most computationally intensive component (per [this documentation](https://colab.research.google.com/drive/10tQ09HK-AQ4-5jIrad4sS176Q_Z0LX6Q?usp=sharing)).

What exactly is being loaded by the `Generate Image` cell that takes so long before actual inference begins? Am I just mistaking this whole "issue" as long initial overhead setup?

### Reproduction

You can reproduce the logs by running all cells in [this Colab notebook](https://colab.research.google.com/drive/14JTBSkKKkvr8BwUb9CH59ZtLGwzprE2T?usp=sharing).

I tested on an A100 GPU only (Colab Pro). If there is a better GPU for my use case, do let me know!

### System Info

- `diffusers` version: 0.27.0
- Platform: Linux-6.1.58+-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.2.1+cu121 (True)
- Huggingface_hub version: 0.20.3
- Transformers version: 4.38.2
- Accelerate version: 0.28.0
- xFormers version: not installed
- Using GPU in script?: <A100 GPU>
- Using distributed or parallel set-up in script?: <fill in>

Colab Pro

### Who can help?

@yiyixuxu @sayakpaul

**Comment by RizzwareEngineer** [comment]
Oh extremely sorry I did not mean to label this as a bug.

**Comment by standardAI** [comment]
The compilation takes place when you run the first inference.

**Comment by sayakpaul** [comment]
As @standardAI pointed out. The first pipeline call will be time-consuming as there's a lot of stuff happening in the background. Second pipeline call should yield faster inference. 

This issue is quite not concerned with the library in general. So, I am going to close this. Feel free to continue the discussion as a discussion thread.
